mptt/filters.py

```diff
@@ -0,0 +1,116 @@
+from django.conf import settings
+from django.contrib.admin import RelatedFieldListFilter
+from django.contrib.admin.options import IncorrectLookupParameters
+from django.contrib.admin.utils import get_model_from_relation
+from django.core.exceptions import ValidationError
+from django.utils.encoding import smart_text
+from django.utils.html import mark_safe
+from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import get_language_bidi
+from django.db.models.fields.related import ForeignObjectRel, ManyToManyField
+
+MPTT_ADMIN_LEVEL_INDENT = getattr(settings, 'MPTT_ADMIN_LEVEL_INDENT', 10)
+
+
+class TreeRelatedFieldListFilter(RelatedFieldListFilter):
+    """
+    Admin filter class which filters models related to parent model with all it's descendants.
+
+     Usage:
+
+    from mptt.filters import TreeRelatedFieldListFilter
+
+    @admin.register(models.MyModel)
+    class MyModelAdmin(admin.ModelAdmin):
+        model = models.MyModel
+        list_filter = (
+            ('my_related_model', TreeRelatedFieldListFilter),
+                    )
+    """
+    template = 'admin/mptt_filter.html'
+
+    # Ripped from contrib.admin.filters,RelatedFieldListFilter Django 1.8 to
+    # change lookup name to 'inhierarchy'
+    def __init__(self, field, request, params, model, model_admin, field_path):
+        self.other_model = get_model_from_relation(field)
+        if hasattr(field, 'rel'):
+            rel_name = field.rel.get_related_field().name
+        else:
+            rel_name = self.other_model._meta.pk.name
+        self.rel_name = rel_name
+        self.lookup_kwarg = '%s__%s__inhierarchy' % (field_path, rel_name)
+        self.lookup_kwarg_isnull = '%s__isnull' % field_path
+        self.lookup_val = request.GET.get(self.lookup_kwarg)
+        self.lookup_val_isnull = request.GET.get(self.lookup_kwarg_isnull)
+        self.lookup_choices = self.field_choices(field, request, model_admin)
+        super(RelatedFieldListFilter, self).__init__(field, request, params,
+                                                     model, model_admin, field_path)
+        if hasattr(field, 'verbose_name'):
+            self.lookup_title = field.verbose_name
+        else:
+            self.lookup_title = self.other_model._meta.verbose_name
+        self.title = self.lookup_title
+
+    # Ripped from contrib.admin.filters,FieldListFilter Django 1.8 to deal with
+    # lookup name 'inhierarchy'
+    def queryset(self, request, queryset):
+        try:
+            # #### MPTT ADDITION START
+            if self.lookup_val:
+                other_model = self.other_model.objects.get(pk=self.lookup_val)
+                other_models = other_model.get_descendants(True)
```

Comment:
Since this is using `get_descendants`, does this mean that when you click on 'ChildMaterial1' the filter will _only_ show 'ChildMaterial1' ?

Would using `get_family` be better? Then it would show 'ParentMaterial' too so you could traverse back up the tree easily.